File: src/main/java/com/nearvanilla/icecream/modules/lightning/LightningModule.java
/* Licensed under GNU Affero General Public License v3.0 */
package com.nearvanilla.icecream.modules.lightning;

import com.nearvanilla.icecream.IceCream;
import com.nearvanilla.icecream.modules.ModuleInterface;
import com.nearvanilla.icecream.modules.lightning.commands.Lightning;
import org.bukkit.event.Listener;

public class LightningModule implements ModuleInterface {

  private boolean isEnabled = false;

  @Override
  public boolean shouldEnable() {
    return IceCream.getInstance().getCustomConfig().getBoolean("modules.lightning");
  }

  @Override
  public void registerEvents(Listener listener) {
    IceCream.getInstance().getLogger().warning("LightningModule does not have any events.");
  }

  @Override
  public void unregisterEvents(Listener listener) {
    IceCream.getInstance().getLogger().warning("LightningModule does not have any events.");
  }

  @Override
  public void registerCommands() {
    IceCream.getInstance().getCommand("lightning").setExecutor(new Lightning());
  }

  @Override
  public void unregisterCommands() {
    IceCream.getInstance().getCommand("lightning").setExecutor(null);
  }

  @Override
  public void enable() {
    if (!isEnabled) {
      registerCommands();
      isEnabled = true;
    }
  }

  @Override
  public void disable() {
    if (isEnabled) {
      unregisterCommands();
      isEnabled = false;
    }
  }
}

File: src/main/java/com/nearvanilla/iceCream/modules/lightning/LightningModule.kt
/* Licensed under the GNU Affero General Public License. */
package com.nearvanilla.iceCream.modules.lightning

import com.nearvanilla.iceCream.IceCream
import com.nearvanilla.iceCream.libs.Module
import com.nearvanilla.iceCream.modules.lightning.commands.StrikeCommand

class LightningModule : Module {
    override fun shouldEnable(): Boolean {
        val enabled = IceCream.pluginInstance.config.getBoolean("modules.lightning.enabled")
        IceCream.pluginInstance.logger.info("Lightning Module enabled: $enabled")
        return enabled
    }

    override fun registerEvents() {}

    override fun unRegisterEvents() {}

    override fun registerCommands() {
        IceCream.annotationParser.parse(StrikeCommand())
    }

    override fun unregisterCommands() {
        IceCream.commandManager.deleteRootCommand("strike")
    }
}
